core/frontend/src/libs/filebrowser.ts

```diff
@@ -0,0 +1,153 @@
+
+  /* Fetch a folder from filebrowser. */
+  /**
+   * @param folder_path - String absolute path of folder to be fetched
+   * @returns FilebrowserFolder object
+  * */
+  async fetchFolder(folder_path: string): Promise<FilebrowserFolder> {
+    return axios({
+      method: 'get',
+      url: `${filebrowser_url}/resources${folder_path}`,
+      timeout: 10000,
+      headers: { 'X-Auth': this.filebrowser_token() },
+    })
+      .then((response) => response.data)
+      .catch((error) => {
+        const error_message = `Could not fetch folder ${folder_path}: ${error.message}`
+        notification_store.pushNotification(new LiveNotification(
+          NotificationLevel.Error,
+          filebrowser_service,
+          'FOLDER_FETCH_FAIL',
+          error_message,
+        ))
+        throw new Error(error_message)
+      })
+  }
+
+  /* Delete a single file. */
+  /* Register a notification and throws if delete fails. */
+  /**
+   * @param file - FilebrowserFile object to be deleted
+  * */
+  async deleteFile(file: FilebrowserFile): Promise<void> {
+    axios({
+      method: 'delete',
+      url: `/file-browser/api/resources${file.path}`,
+      timeout: 10000,
+      headers: { 'X-Auth': this.filebrowser_token() },
+    })
+      .catch((error) => {
+        const error_message = `Could not delete file ${file.path}: ${error.message}`
+        notification_store.pushNotification(new LiveNotification(
+          NotificationLevel.Error,
+          filebrowser_service,
+          'FILE_DELETE_FAIL',
+          error_message,
+        ))
+        throw new Error(error_message)
+      })
+  }
+
+  /* Performs multiple-file delete requests. */
+  /* Will perform all possible delete operations and reject if any of them fails. */
+  /**
+   * @param files - FilebrowserFile objects to be deleted
+  * */
+  async deleteFiles(files: FilebrowserFile[]): Promise<void> {
+    const delete_promises: Promise<void>[] = []
+    files.forEach((file) => {
+      delete_promises.push(this.deleteFile(file))
+    })
+    await Promise.all(delete_promises)
+  }
+
+  /* Returns the relative URL (without hostname) of a single file. */
+  /**
+   * @param file - FilebrowserFile object
+  * */
+  singleFileRelativeURL(file: FilebrowserFile): string {
+    return `${filebrowser_url}/raw${file.path}?auth=${this.filebrowser_token()}`
+  }
+
+  /* Returns the relative URL (without hostname) of a zip of multiple files. */
+  /**
+   * @param files - FilebrowserFile objects
+  * */
+  multipleFilesRelativeURL(files: FilebrowserFile[]): string {
+    const files_arg = files.map((file) => file.path).join(',')
+    return `${filebrowser_url}/raw/?files=${files_arg}&algo=zip&auth=${this.filebrowser_token()}`
+  }
+
+  /* Download files (single or multiple). */
+  /**
+   * @param files - FilebrowserFile objects array
+  * */
+  downloadFiles(files: FilebrowserFile[]): void {
+    let url = ''
+    if (files.length === 1) {
+      url = this.singleFileRelativeURL(files[0])
```

Comment:
at a first glance, this looks unnecessary. isn't `multipleFilesRelativeUrl` able to handle a list with only one element?

Reply:
oh, it zips all the files together when it's multiple files? then I agree it is best to leave it as is.

Reply:
> oh, it zips all the files together when it's multiple files? then I agree it is best to leave it as is.

Yup